src/components/services-view/service-view.classes.ts

```diff
@@ -0,0 +1,181 @@
+import {allCategories} from '../../constants';
+import * as _ from 'lodash';
+
+interface IServiceItem {
+    icon: string;
+    name: string;
+    category: string;
+    subCategory: string;
+    description: string;
+    longDescription: string;
+    data: any;
+}
+
+export class CatalogServiceItem implements IServiceItem {
+    public icon: string;
+    public name: string;
+    public category: string;
+    public subCategory: string;
+    public description: string;
+    public longDescription: string;
+    public data: any;
+
+    constructor (catalogService: any) {
+        this.data = catalogService;
+        this.icon = this.getIcon();
+        this.name = this.getName();
+        this.description = this.getDescription();
+        this.longDescription = this.getLongDescription();
+        this.subCategory = this.getSubCategory();
+        this.category = this.getCategory();
+    }
+
+    private getIcon() {
+        return this.data.imageURL;
+    }
+
+    private getName() {
+        return this.data.displayName ? this.data.displayName : this.data.metadata.name;
+    }
+
+    private getDescription() {
+        return this.data.description;
+    }
+
+    private getLongDescription() {
+        return this.data.longDescription;
+    }
+
+    private getSubCategory() {
+        if (this.data.spec && this.data.spec.tags) {
+            let lastTag = this.data.spec.tags.pop();
+            return getSubCategoryFromAlias(lastTag);
+        }
+    }
+
+    private getCategory() {
+        return getCategoryFromSubCategory(this.subCategory);
+    }
+}
+
+export class OpenshiftImageItem implements IServiceItem {
+    public icon: string;
+    public name: string;
+    public category: string;
+    public subCategory: string;
+    public description: string;
+    public longDescription: string;
+    public data: any;
+    public isBuilder: boolean;
+    private annotations: any;
+
+    constructor (openshiftImage: any) {
+        this.data = openshiftImage;
+        this.annotations = this.getAnnotations();
+        this.icon = this.getIcon();
+        this.name = this.getName();
+        this.description = this.getDescription();
+        this.longDescription = this.getLongDescription();
+        this.subCategory = this.getSubCategory();
+        this.category = this.getCategory();
+    }
+
+    private getAnnotations() {
+        if (this.data.spec && this.data.spec.tags && this.data.spec.tags.length > 1 &&
+            this.data.spec.tags[0].annotations) {
+            return this.data.spec.tags[0].annotations;
+        }
+    }
+
+    private getIcon() {
+        if (this.annotations) {
+            return 'font-icon ' + this.annotations.iconClass;
+        }
+    }
+
+    private getName() {
+        if (this.data.metadata.annotations) {
+            return this.data.metadata.annotations['openshift.io/display-name'];
```

Comment:
I noticed @jeff-phillips-18 added this filter in https://github.com/openshift/origin-web-common/pull/8